server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4758,11 +4767,21 @@ private String copySegmentFiles(
                 }
             }
             for (String file : uploadedSegments.keySet()) {
+                RemoteSegmentStoreDirectory.UploadedSegmentMetadata segmentMetadata = uploadedSegments.get(file);
                 long checksum = Long.parseLong(uploadedSegments.get(file).getChecksum());
                 if (overrideLocal || localDirectoryContains(storeDirectory, file, checksum) == false) {
-                    storeDirectory.copyFrom(sourceRemoteDirectory, file, file, IOContext.DEFAULT);
-                    storeDirectory.sync(Collections.singleton(file));
-                    downloadedSegments.add(file);
+                    long startTimeInNs = System.nanoTime();
```

Comment:
We can use System.currentMillis as this is only used for stats currently. If required for any critical purpose, we will move to System.nanos.